os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -547,6 +547,9 @@ void binary_manager_release_binary_sem(int bin_idx)
 	flags = irqsave();
 
 	sem = (sem_t *)sq_peek(&g_sem_list);
+	if (sem == NULL) {
+		bmdbg("g_sem_list is empty.\n");
+	}
 	while (sem) {
```

Comment:
How about adding `else` condition to prevent duplicated checking when it is null?